### DISCUSSION TITLE
When to use reporting in device configure?

### DISCUSSION THREAD

**Discussion by pearson** [Koenkk/zigbee2mqtt#13107] [General]
I'm a bit confused about when certain configure statements are used in converters. For example, with on/off devices,  some of the existing converters will have two lines in their configure statement such as:
```javascript
await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff']);
await reporting.onOff(endpoint);
```

Some other examples will have the `reporting.bind(...)` line, but not have the `reporting.onOff(...)` line.

Why is that, and what's the case where this works without the `reporting.onOff(...)` line? Is it simply a matter of not wanting to receive updates regarding the device's on/off status?

Thanks!

**Comment by Koenkk** [comment]
Some devices already start reporting their state without the `reporting.onOff()` thus it is not needed. (it varies per device)

**Reply by pearson** [reply]
Ah, I see.

At first read through those configuration files, it's a bit of a mystery as to why some statements are required for some devices and not others. 

Thanks!
